pkg/pgmodel/ingestor/inserter.go

```diff
@@ -86,6 +94,42 @@ func (p *pgxInserter) runCompleteMetricCreationWorker() {
 	}
 }
 
+func (p *pgxInserter) runSeriesEpochSync() {
+	epoch, err := p.refreshSeriesEpoch(model.InvalidSeriesEpoch)
+	// we don't have any great place to report errors, and if the
+	// connection recovers we can still make progress, so we'll just log it
+	// and continue execution
+	log.Error("msg", "error refreshing the series cache", "err", err)
+	for range p.seriesEpochRefresh.C {
+		epoch, err = p.refreshSeriesEpoch(epoch)
+		log.Error("msg", "error refreshing the series cache", "err", err)
+	}
+}
+
+func (p *pgxInserter) refreshSeriesEpoch(existingEpoch model.SeriesEpoch) (model.SeriesEpoch, error) {
+	dbEpoch, err := p.getServerEpoch()
+	if err != nil {
+		// Trash the cache just in case an epoch change occurred, seems safer
+		p.scache.Reset()
```

Comment:
Is it a good idea to store the epoch with each series so we can purge only data from an earlier epoch?

Reply:
Good idea but I think separate PR